storage.go

```diff
@@ -25,13 +25,38 @@ const (
 )
 
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	o = s.newObject(false)
-	o.Mode = ModeRead
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(true)
+		o.Mode = ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode = ModeRead
+	}
+
 	o.ID = s.getAbsPath(path)
 	o.Path = path
 	return o
 }
 
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	err = s.bucket.Mkdir(rp)
+	if err != nil && checkErrorCode(err, responseCodeFolderAlreadyExist) {
+		// Omit `folder already exists` error here.
+		err = nil
+	}
+	if err != nil {
+		return
+	}
+
+	o = s.newObject(true)
```

Comment:
If the dir has been created, we need to use `s.newObject(false)` to allow the user to fetch data from the remote.